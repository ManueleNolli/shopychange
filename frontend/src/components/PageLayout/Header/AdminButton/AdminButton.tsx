import React from 'react'

import { Box, Button } from '@chakra-ui/react'
import { FaUnlock } from 'react-icons/fa'
import { useNavigate } from 'react-router-dom'
import AdminCheck from '../../../../utils/AdminCheck/AdminCheck'

export default function AdminButton() {
  const navigate = useNavigate()

  const onNavigation = () => {
    navigate('/admin')
  }

  return (
    <AdminCheck redirectHome={false}>
      <Box>
        <Button
          data-testid="admin-button"
          aria-label="Admin"
          _focus={{ boxShadow: 'none' }}
          w="fit-content"
          borderColor={'red.500'}
          borderWidth={'2px'}
          onClick={() => onNavigation()}
        >
          <FaUnlock />
        </Button>
      </Box>
    </AdminCheck>
  )
}
