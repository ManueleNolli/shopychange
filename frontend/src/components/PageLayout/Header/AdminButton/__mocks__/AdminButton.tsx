import { divWithoutChildrenMock } from '../../../../../../mocksUtils'

const AdminButtonMock = () => divWithoutChildrenMock('AdminButtonContent')
export default AdminButtonMock
