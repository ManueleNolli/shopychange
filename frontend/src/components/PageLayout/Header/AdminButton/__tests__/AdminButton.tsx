import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import AdminButton from '../AdminButton'

const mockNavigate = jest.fn()
jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
}))

jest.mock('../../../../../utils/AdminCheck/AdminCheck', () => ({
  __esModule: true,
  default: ({ children }: { children: React.ReactNode }) => (
    <div data-testid={'AdminCheck'}>{children}</div>
  ),
}))

describe('AdminButton', () => {
  test('Render', () => {
    const tree = render(<AdminButton />)
    expect(tree.getByTestId('admin-button')).toBeInTheDocument()
    expect(tree.container).toMatchSnapshot()
  })
  test('Navigate', () => {
    const tree = render(<AdminButton />)
    const button = tree.getByTestId('admin-button')
    fireEvent.click(button)
    expect(mockNavigate).toBeCalledTimes(1)
    expect(mockNavigate).toBeCalledWith('/admin')
  })
})
