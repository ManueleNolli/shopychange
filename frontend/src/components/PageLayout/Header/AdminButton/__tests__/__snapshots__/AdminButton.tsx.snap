// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminButton Render 1`] = `
<div>
  <div
    data-testid="AdminCheck"
  >
    <div
      class="css-0"
    >
      <button
        aria-label="Admin"
        class="chakra-button css-1wnmm18"
        data-testid="admin-button"
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 256H152V152.9c0-39.6 31.7-72.5 71.3-72.9 40-.4 72.7 32.1 72.7 72v16c0 13.3 10.7 24 24 24h32c13.3 0 24-10.7 24-24v-16C376 68 307.5-.3 223.5 0 139.5.3 72 69.5 72 153.5V256H48c-26.5 0-48 21.5-48 48v160c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V304c0-26.5-21.5-48-48-48z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
